### PR TITLE
[bitnami/contour] Add the possibility to set HostIP

### PR DESCRIPTION
**Pull request by bokysan**
**Description of the change**

The cart does not allow setting the `hostIP` parameter on the
Envoy `DeamonSet`. This is useful in certain cases (e.g. with `k3s`
and single-node deployments) and should be possible.

This commit simply adds the possibility to set the `hostIP` on ports
exposed by Envoy.

**Benefits**

It's now able to run two Contour servers on the same host and same
ports with different IP addresses (e.g. for public and private IPs).

**Possible drawbacks**

None

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] ~Is the PR adding a new container? Please reviewer, add it to the models (internal process)~
- [X] Is the PR adding a new parameter? Please, ensure it’s documented in the README.md

### CODE REVIEW

File: bitnami/contour/README.md [hunk 1]
```diff
@@ -174,6 +174,10 @@ The following tables lists the configurable parameters of the contour chart and
...
 | `envoy.hostPorts.http`                     | Sets `hostPort` http port                                                                                                                                                                                    | `80`                                                    |
 | `envoy.hostPorts.https`                    | Sets `hostPort` https port                                                                                                                                                                                   | `443`                                                   |
+| `envoy.useHostIP`                          | Enable/disable `hostIP`                                                                                                                                                             | `true`                                                  |
```
Comment (alemorcuq) [inline bitnami/contour/README.md:177]:
```suggestion
| `envoy.useHostIP`                          | Enable/disable `hostIP`                                                                                                                                                             | `false`                                                  |
```

File: bitnami/contour/templates/envoy/daemonset.yaml [hunk 2]
```diff
@@ -146,12 +146,18 @@ spec:
...
+              hostIP: {{ .Values.envoy.hostIPs.http }}
+              {{- end }}
               name: http
               protocol: TCP
             - containerPort: {{ .Values.envoy.containerPorts.https }}
               {{- if .Values.envoy.useHostPort }}
               hostPort: {{ .Values.envoy.hostPorts.https }}
               {{- end }}
+              {{- if .Values.envoy.useHostIP }}
+              hostIP: {{ .Values.envoy.hostIPs.http }}
```
Comment (alemorcuq) [inline bitnami/contour/templates/envoy/daemonset.yaml:159]:
```suggestion
              hostIP: {{ .Values.envoy.hostIPs.https }}
```

File: bitnami/contour/values.yaml [hunk 3]
```diff
@@ -515,13 +515,23 @@ envoy:
...
+  ##
+  useHostIP: false
 
   ## HTTP and HTTPS ports
   ##
   hostPorts:
     http: 80
     https: 443
 
+  ## HTTP and HTTPS ports
```
Comment (alemorcuq) [inline bitnami/contour/values.yaml:528]:
```suggestion
  ## HTTP and HTTPS IPs
```

**Review by alemorcuq — CHANGES_REQUESTED** [review]
Thanks for the PR! Please, see my suggestions below.

**Comment by bokysan** [comment]
Changes accepted. Branch rebased.

**Review by alemorcuq — APPROVED** [review]
Thanks

**Pull request merged**